### DISCUSSION TITLE
Yuzu shutdown, Box86/Box64 compatability, and native Linux game compatability...

### DISCUSSION THREAD

**Discussion by Tell-Sackett** [ruanformigoni/gameimage#43] [Ideas]
I have a notice and two questions:

1. Recently, Yuzu was copyright struck by Nintendo and its repos, wiki, etc. taken down. It might be a good idea to swap over to Ryujinx as the Switch emulator used in packaging. Currently, a forked version of Yuzu, known as "Suyu" is available, but its developmental future is unproven; Ryujinx has been around for, basically, as long as Yuzu was and has proven itself to be enduring and fully-featured. I would look at contributing a change myself, but I'm currently no developer.

2. Would it be possible to make a Box86/Box64 option for packaging such that packaged games could be run on ARM devices?

3. Perhaps this option is unnecessary due to the availability of packaging one's own flatimage, but for completeness' sake of gameimage as a tool: are there any plans to provide an option for native Linux game packaging?

Thanks for your time and this cool application!

**Comment by ruanformigoni** [comment]
Hello @Tell-Sackett ! Thanks for the heads up, I had removed the yuzu repo and now have update the README.md to avoid confusion.

I will indeed replace yuzu with Ryujinx, currently I'm reworking the project from scratch to be user-friendly and completely GUI oriented (should be done in a couple of weeks).

Running on ARM is possible, I still have to implement that on flatimage: https://github.com/ruanformigoni/flatimage/issues/16

Yes, I do plan to include linux packaging, I just did not include it yet because of some issues I had with AppImage, it seems to work fine now on FlatImage, so I'll add it as an option :smile: 

Any other questions feel free to ask!

**Reply by Tell-Sackett** [reply]
Hello @ruanformigoni!

1. Great to hear! Ngl, I was kinda' getting worried about the tool for a while there (I'm still relatively new to GitHub/GitLab and am not familiar with time/workflow norms). That said, after seeing the other day that you were still actively working on FlatImage in a seperate repo, my worries were settled a little. This is a really cool tool, imo; I'm hoping to, at some point, package a whole modded Skyrim into a single bundle!

From the sounds of it, this FlatImage packaging system sounds really handy, hope it catches on!

2. Thanks for the redirect. I should have thought to look on the FlatImage repo for info regarding ARM support.

3. Awesome!

Really appreciate the info!

**Reply by ruanformigoni** [reply]
No problem! You can follow the progress on development on the [cpp branch](https://github.com/ruanformigoni/gameimage/tree/cpp). If you'd like to try it out, [here](https://mega.nz/file/B8BgCagI#-JCEDX0V-yox3d5yWhrielbx8FyUqSn9d65Tihosjhs) is a preview version of it, I'd greatly appreciate your feedback on the user-experience.

Modding will come after native-linux and Ryujinx are integrated, flatimage uses overlayfs, so you can have a folder with all your mods in the dwarfs format and ask gameimage to overlay them over the data/ folder of the skyrim wine installation. This would allow you to use different mod packs with the same skyrim gameimage.

**Reply by Tell-Sackett** [reply]
Cool, I didn't even notice the cpp branch; I've been busy with lots of life stuff lately, but I'll see what I can do!

Interesting, I'll have to do more research on this overlayfs stuff then. I had just been planning on trying to shove MO2 into the file structure in-between packaging steps 😅 (didn't actually make it around to trying it yet though). Your's sounds like it might become a much easier solution.

**Comment by ruanformigoni** [comment]
Project rework released, ryujinx is presenting problems, so it is not yet available. However, linux native is working :smile:

**Comment by Tell-Sackett** [comment]
Really cool to see soo much progress being made! Sorry for the late response(s).
     
I just tried running v1.1.0 to make both a native Linux FI and wine FI; however, I failed to get either to work.

For the native Linux game, I had two problems. First, I was having problems on the "Fetch the Required Files" screen. The fetching process would seemingly finish (the terminal would print "Download result: Ok(())", but strangely enough only part 1 of linux.tar.xz seems to have been fetched, there is no reference to completing part 2 or 3 in the terminal), but the "next" button would do nothing when pressed (this process was seemingly resolved after exiting the application, rearranging directory structures, and renaming some files).

The second problem with the Linux native game, which led to me giving up, was when I reached the "[Select a Binary]" screen. This screen was blank for me, and anytime I tried to hit the "Next" button, the terminal window would print something like the following "[No button selected]". I could not find away around this (I even clicked around on the blank screen in hopes that there were buttons that just weren't visible).

The problem I had with the Wine game was as follows: I would reach the "Fetch the Required Files" screen, where I would be prompted to fetch two different packages. After both of these would complete, I would hit "Next", and, while the terminal would print some things before stopping, I would still be on the same page. The terminal print ended with something like "[File list not found]" or something.

If you need any more details, I can try to rerun these tests and provide more info. Also, If I need to move this conversation somewhere else (other than this discussion page) just let me know.

**Comment by ruanformigoni** [comment]
Hi @Tell-Sackett ! Sorry for the troubles, I fixed a bug on `v1.1.0` where the download would "give out" if it took longer than expected, which should resolve your issues with that.

> The second problem with the Linux native game, which led to me giving up, was when I reached the "[Select a Binary]" screen. This screen was blank for me, and anytime I tried to hit the "Next" button, the terminal window would print something like the following "[No button selected]". I could not find away around this (I even clicked around on the blank screen in hopes that there were buttons that just weren't visible).

This screen shows the `.sh` files installed, gameimage now will warn you if it cannot find a file to select instead of drawing a blank screen. I have created a tutorial for linux:

https://github.com/ruanformigoni/gameimage/assets/18180693/bd74f416-0ea2-45be-9382-7beed0d8a682

Thanks for the feedback!

**Reply by Tell-Sackett** [reply]
Ahh, I see; that's my bad, I should have been more broad and thorough with the game(s) I tested. I was doing two things different:

1) The game I was attempting to package has the extension ".x86_64". To be precise, the game I was attempting to package was "Buckshot Roulette" which is distributed in the form of a single executable file (Buckshot Roulette.x86_64). For reference, it can be found at the following itch.io page: https://mikeklubnika.itch.io/buckshot-roulette. Also, just for context, I read that there are other Linux games which have this extension (they are probably _also_ games which are distributed through itch.io, as that site more often has developers distributing their games as raw executables/files), and I know of at least one other application which uses this file extension (that being the Godot game engine).

2) The native Linux game I was attempting to package was not distributed in the form of a GOG installer. I attempted two different methods of getting it to process: (1) I tried to select the executable in the "Install the Application" step (by changing file types to "all files") and when that didn't work (2) I did not select anything on the "Install the Application" step (with the intention of selecting the game executable on the "Select the Main Binary" page.

I will look at testing out the program some more tonight to see if I can get GOG native Linux games packaging smoothly (as well, perhaps, as some emulator and wine games).
